src/octopeer-github/test/ElementSelectionBehaviour/SelectButtonsElementSelectionBehaviourTest.ts

```diff
@@ -1,26 +1,153 @@
+    [LabelSettingElementSelectionBehaviour, 1],
+    [LockConversationSettingElementSelectionBehaviour, 1],
+    [MilestoneSettingElementSelectionBehaviour, 1],
+    [UnsubscribeSettingElementSelectionBehaviour, 1],
+    [CommitsTabHeaderElementSelectionBehaviour, 1],
+    [ConversationTabHeaderElementSelectionBehaviour, 1],
+    [FilesChangedTabHeaderElementSelectionBehaviour, 1],
+];
+
+let commitSelectors: CreatableNumberPair[] = [
+    [CancelEditPRNameButtonElementSelectionBehaviour, 0],
+    [EditPRNameButtonElementSelectionBehaviour, 1],
+    [SaveEditPRNameButtonElementSelectionBehaviour, 0],
+    [DateMiscellaneousElementSelectionBehaviour, 4],
+    [CommitHashcodeNameElementSelectionBehaviour, 4],
+    [CommitMessageNameElementSelectionBehaviour, 4],
+    // [OtherContributerNameElementSelectionBehaviour, 0],
+    // [PRCreatorNameElementSelectionBehaviour, 4],
+    // [PRParticipantNameElementSelectionBehaviour, 0],
+    [CommitsTabHeaderElementSelectionBehaviour, 1],
+    [ConversationTabHeaderElementSelectionBehaviour, 1],
+    [FilesChangedTabHeaderElementSelectionBehaviour, 1],
+];
+
+let filesChangedSelectors: CreatableNumberPair[] = [
+    [AddEmoticonButtonElementSelectionBehaviour, 1],
+    [CancelEditPRNameButtonElementSelectionBehaviour, 0],
+    // [CancelInlineCommentButtonElementSelectionBehaviour, 0],
+    // [CommentInlineCommentButtonElementSelectionBehaviour, 0],
+    [EditCommentButtonElementSelectionBehaviour, 1],
+    [EditPRNameButtonElementSelectionBehaviour, 1],
+    // [InlineCommentButtonElementSelectionBehaviour, 0],
+    [SaveEditPRNameButtonElementSelectionBehaviour, 0],
+    [DateMiscellaneousElementSelectionBehaviour, 5],
+    // [OtherContributerNameElementSelectionBehaviour, 0],
+    // [PRCreatorNameElementSelectionBehaviour, 0],
+    // [PRParticipantNameElementSelectionBehaviour, 1],
+    [CommitsTabHeaderElementSelectionBehaviour, 1],
+    [ConversationTabHeaderElementSelectionBehaviour, 1],
+    [FilesChangedTabHeaderElementSelectionBehaviour, 1],
+];
+
+const htmlsList = [
+    "Conversation_tab.html",
+    "Commit_tab.html",
+    "Files_changed_tab.html",
+];
+
+const selectorListsList = [
+    conversationSelectors,
+    commitSelectors,
+    filesChangedSelectors,
+];
+
+function functionName(fun: ElementSelectionBehaviourCreatable) {
+    let ret = fun.toString();
+    ret = ret.substr("function ".length);
+    ret = ret.substr(0, ret.indexOf("("));
+    return ret;
+}
+
+let databaseSpy: jasmine.Spy;
+jasmine.getFixtures().fixturesPath = "base/src/octopeer-github/test/resources";
+
+for (let i = 0; i < htmlsList.length; i++) {
+    const htmlFile = htmlsList[i];
+    const selectors = selectorListsList[i];
+
+    // console.log(selectors);
+    for (let j = 0; j < selectors.length; j++) {
+        const selectorCreatable = selectors[j][0];
+        describe("An ElementSelector that selects Buttons " + functionName(selectorCreatable) + " at " + htmlFile, function () {
```

Comment:
The describe name is not properly formatted. It now says: 
`An ElementSelector that selects Buttons MergePRButtonElementSelectionBehaviour at Commit_tab.html`
while it should be:
`A MergePRButtonElementSelectionBehaviour that selects Elements at Commit_tab.html`